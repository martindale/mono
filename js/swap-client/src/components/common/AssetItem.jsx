import { Grid, IconButton, Stack } from '@mui/material'
import React from 'react'
import NavigateNextIcon from '@mui/icons-material/NavigateNext'

export const AssetItem = ({ asset, handleClick }) => {
  return (
    <Grid item container direction='row' onClick={e => handleClick(asset)} style={{ cursor: 'pointer' }} className='asset-item'>
      <Grid item xs={1.5}><img width='32px' src={asset.img_url} /></Grid>
      <Grid item xs={5} container direction='column' textAlign='left'>
        <span><h4>{asset.title}</h4></span>
        {asset.type && <span><h5 style={{ fontSize: '0.8em', color: 'grey' }}>{asset.type}</h5></span>}
      </Grid>
      <Grid item xs={5.5} textAlign='right' alignItems='center'>
        {asset.amount}<span style={{ color: 'grey', fontSize: '0.7em' }}>{asset.type && asset.type.toUpperCase()}</span><NavigateNextIcon />
      </Grid>
    </Grid>
  )
}
