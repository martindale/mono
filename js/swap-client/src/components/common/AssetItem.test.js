import { AssetItem } from './AssetItem'
import { create, act } from 'react-test-renderer'

describe('AccessOptionComponent', () => {
  it('should work', () => {
    let tree
    const asset = {
      img_url: '123',
      title: 'card',
      type: '123',
      amount: 1
    }
    const handleClick = jest.fn()

    act(() => {
      tree = create(
        <AssetItem
          asset={asset}
          handleClick={handleClick}
        />
      )
    })

    expect(tree).toMatchSnapshot()
  })
  afterAll(() => jest.resetModules())
})
