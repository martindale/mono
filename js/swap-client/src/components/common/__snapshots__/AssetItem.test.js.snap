// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessOptionComponent should work 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  onClick={[Function]}
  style={
    {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1.5 css-1duidt6-MuiGrid-root"
  >
    <img
      src="123"
      width="32px"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-5 css-12dbh09-MuiGrid-root"
  >
    <span>
      <h4>
        card
      </h4>
    </span>
    <span>
      <h5
        style={
          {
            "color": "grey",
            "fontSize": "0.8em",
          }
        }
      >
        123
      </h5>
    </span>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5.5 css-1c3ja2f-MuiGrid-root"
  >
    1
    <span
      style={
        {
          "color": "grey",
          "fontSize": "0.7em",
        }
      }
    >
      123
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="NavigateNextIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
</div>
`;
